packages/esm-patient-chart-app/src/visit/visits-widget/past-visits-components/encounters-table/all-encounters-table.component.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useMemo, useState } from 'react';
+import EncountersTable from './encounters-table.component';
+import { type EncountersTableProps, mapEncounter, usePaginatedEncounters } from './encounters-table.resource';
+import { type EncounterType } from '@openmrs/esm-framework';
+
+interface AllEncountersTableProps {
+  patientUuid: string;
+}
+
+/**
+ * This component shows a table of all encounters (across all visits) of a patient
+ */
+const AllEncountersTable: React.FC<AllEncountersTableProps> = ({ patientUuid }) => {
+  const pageSize = 20;
```

Comment:
Presumably at some point this will become somehow configurable? In general, doesn't it make sense to have this be a property controlled by the datatable itself, i.e., via something like the "Items per page:" bit [here](https://carbondesignsystem.com/components/data-table/usage/#anatomy)?